### DISCUSSION TITLE
"flask create-app app-name" script

### DISCUSSION THREAD

**Discussion by dilrajs616** [pallets/flask#5619] [Ideas]
I created a script called create-app. It can be used by running flask "create-app demo-name" in the terminal. It creates a simple react app with following directory structure:
static
   ├───css
   ├───img
   └───js
templates
app.py
config.py
.gitignore
README.md
requirements.txt

templates folder has index.html file and css folder has index.css file in it. app.py simply renders index.html. this script also initialises git. It will be a very nice way of setting up flask apps by a single command. I want to contribute this to official flask repo. How can I do that?

**Comment by tombohub** [comment]
how do we use this script?

**Reply by dilrajs616** [reply]
I made a pip package called "flask-make". you can install it by using "pip install flask-make". then for using it you can simply use the command "flask-make my-app" this will create a sample app in a directory named "my-app". There will be a virtual environment in it as well. You can directly start making modules in this new app

**Reply by dilrajs616** [reply]
https://pypi.org/project/flask-make/

here is the documentation if you are interested

**Comment by CheeseCake87** [comment]
This might fit better as a Flask extension.

**Reply by ThiefMaster** [reply]
or even as something completely standalone... why would you use that instead of flask-cookiecutter though? (no idea how well that one is maintained)

**Reply by dilrajs616** [reply]
I published it as a pip package. It can be installed globally and used whenever you need to setup a flask app. The name of the package is "flask-make". You can find all the useful info related to this package here: https://pypi.org/project/flask-make/

**Reply by ThiefMaster** [reply]
Just FYI, there are much better things than a requirements.txt populated from `pip` freeze. Look at pyproject.toml and tools like uv. :)

**Reply by loadept** [reply]
It might be a good idea
